### PR TITLE
Fix ratelimit policy

### PR DESCRIPTION
**Pull request by Plerx2493**
# Summary
Tries to get header and add a check for global limit

### CODE REVIEW

File: DSharpPlus/Net/Rest/RateLimitPolicy.cs [hunk 1]
```diff
@@ -121,19 +123,22 @@ public RateLimitPolicy(ILogger logger)
...
 
-        hash = response.Headers.GetValues("X-RateLimit-Bucket").SingleOrDefault() ?? "UNKNOWN";
-
-        if(!RateLimitBucket.TryExtractRateLimitBucket(response.Headers, out RateLimitBucket? extracted))
+        if (!exemptFromGlobalLimit)
         {
-            return response;
-        }
+            bool hasBucketHeader = response.Headers.TryGetValues("X-RateLimit-Bucket", out IEnumerable<string>? hashHeader);
+            hash = hasBucketHeader ? hashHeader?.FirstOrDefault() : "UNKNOWN";
```
Comment (akiraveliara) [inline DSharpPlus/Net/Rest/RateLimitPolicy.cs:129]: if we don't have a header, dont try to add it to the known hashes. this should just be a `.Single()` call after establishing it exists

**Review by akiraveliara — CHANGES_REQUESTED** [review]
this won't solve it either. also, can you test this or does it return the header now

File: DSharpPlus/Net/Rest/RateLimitPolicy.cs (hunk 1 above)
Comment (Plerx2493) [reply DSharpPlus/Net/Rest/RateLimitPolicy.cs]: should be solved

**Review by akiraveliara — APPROVED** [review]
what about the testing part?

**Comment by Plerx2493** [comment]
tested and works

**Pull request merged**